### PR TITLE
[UDC-121] Add parametric three stripes mobile button to MSUXF mixins

### PR DESCRIPTION
**Pull request by ms-mrigoli**
## Background

_Working in the landing for Compas I realized that the burguer-ish button used in mobile was a bit tricky to make fast in a custom size shape and animated, so I've made a mixin for this purpose and I thought it would be nice to add it to the framework._

## Changes done

- Added mixin c-mobile-menu-button().
- Added markup for documentation.

## Pending to be done
- Maybe we can add some other animations for the transformation.

## Notes
_N/A_

## Screenshots

mobile-menu-button

### CODE REVIEW

File: assets/styles/modules/_buttons.scss [hunk 1]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+///   }
+/// @example markup - For the button to work as intended, the element must have three spans within.
+/// <div class="mobile-button">
+///   <span></span>
+///   <span></span>
+///   <span></span>
+/// </div>
+/// @todo Maybe adding different animations?
+
+@mixin c-mobile-menu-button($width, $height, $stroke, $color, $transition: '.25s ease-in-out') {
```
Comment (lcalvino) [inline assets/styles/modules/_buttons.scss]: For the $transition var you can use the `$msuxf-transition-duration` in order to be consisten with the framework transitions timing, indeeed it has the same value :)

File: assets/styles/modules/_buttons.scss [hunk 2]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+    &:nth-child(3) {
+      bottom: 0;
+      transform-origin: left center;
+    }
+  }
+  &.opened {
+    span {
+      &:nth-child(1) {
+        transform: rotate(45deg);
+        top: -$offset-y;
```
Comment (lcalvino) [inline assets/styles/modules/_buttons.scss]: can you use `-($offset-y)` in order to be more legible?

File: assets/styles/modules/_buttons.scss [hunk 3]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+/// @example markup - For the button to work as intended, the element must have three spans within.
+/// <div class="mobile-button">
+///   <span></span>
+///   <span></span>
+///   <span></span>
+/// </div>
+/// @todo Maybe adding different animations?
+
+@mixin c-mobile-menu-button($width, $height, $stroke, $color, $transition: '.25s ease-in-out') {
+  $sin45: 0.70710678118; //sqrt(2)/2 = sin(45 deg)
```
Comment (lcalvino) [inline assets/styles/modules/_buttons.scss]: such an alchemy!

File: assets/styles/modules/_buttons.scss (hunk 2 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Sure.

File: assets/styles/modules/_buttons.scss (hunk 3 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Magic!

File: assets/styles/modules/_buttons.scss (hunk 1 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Nice catch! Now the default animation is whatever is set in $msuxf-transition-duration and $msuxf-transition-timing.

File: assets/styles/modules/_buttons.scss [hunk 4]
```diff
@@ -51,4 +51,86 @@ $c-button-border-radius: toem(5px) !default;
...
+    &:nth-child(2) {
+      top: calc(50% - #{$stroke} / 2);
+      transform-origin: left center;
+    }
+    &:nth-child(3) {
+      bottom: 0;
+      transform-origin: left center;
+    }
+  }
+  &.opened {
```
Comment (arzafran) [inline assets/styles/modules/_buttons.scss]: SUPER ULTRA MEGA MINOR: 
I'm not sure about the `opened` state... Maybe `active` is a more generic, and widely adopted usage?

File: assets/styles/modules/_buttons.scss (hunk 3 above)
Comment (arzafran) [reply assets/styles/modules/_buttons.scss]: Mathemagic!

File: assets/styles/modules/_buttons.scss (hunk 4 above)
Comment (nazarenooviedo) [inline assets/styles/modules/_buttons.scss]: Please check the naming convention used in other modules, for example into Modals to indicate the "opened" state we use `&--is-visible`.

File: assets/styles/modules/_buttons.scss [hunk 5]
```diff
@@ -51,4 +51,86 @@ $c-button-border-radius: toem(5px) !default;
...
+    height: $stroke;
+    width: 100%;
+    background: $color;
+    border-radius: calc(#{$stroke} / 2);
+    opacity: 1;
+    left: 0;
+    transform: rotate(0deg);
+    transition: #{$transition};
+
+    &:nth-child(1) {
```
Comment (nazarenooviedo) [inline assets/styles/modules/_buttons.scss]: You are repeating the same property with same values for three elements. Please make a refactor to: `&:nth-child(1), &:nth-child(2), &:nth-child(3)`.

File: assets/styles/modules/_buttons.scss [hunk 6]
```diff
@@ -51,4 +51,86 @@ $c-button-border-radius: toem(5px) !default;
...
+///   }
+/// @example markup - For the button to work as intended, the element must have three spans within.
+/// <div class="mobile-button">
+///   <span></span>
+///   <span></span>
+///   <span></span>
+/// </div>
+/// @todo Maybe adding different animations?
+
+@mixin c-mobile-menu-button($width, $height, $stroke, $color, $transition: 'default') {
```
Comment (nazarenooviedo) [inline assets/styles/modules/_buttons.scss]: Please check in other Module (modals for example) how we use it to instantiate them. We use `@include c-modal()` without have a parent class.

File: assets/styles/modules/_buttons.scss (hunk 4 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Done!

File: assets/styles/modules/_buttons.scss [hunk 7]
```diff
@@ -51,4 +51,86 @@ $c-button-border-radius: toem(5px) !default;
       font-size: 120%;
     }
   }
-}
\ No newline at end of file
+}
+
+/// Mixin for generating the burguer-ish button used in mobile menus. This button is also transformed into an X when adding the <code>active</code> class to the element (as an indicator for closing the mobile menu).
```
Comment (nazarenooviedo) [inline assets/styles/modules/_buttons.scss]: Please check in other Module (models for example) how we use a local variables for each module.

File: assets/styles/modules/_buttons.scss (hunk 6 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Modified. I do not agree with this though, in my opinion loses one of the mixins better qualities that is reutilization. Generally the idea of using mixins is to apply a structure of properties to different elements instead of an include-once only code (https://sass-guidelin.es/#mixins), bit if this is already set with all mixins then I guess is ok to follow the same guides.

A pity in my opinion though, since this button may be nice to use as a general nav menu button and also as an options menu for a widget. This modification makes it less dynamic since only applies this properties to a single hardcoded class.

File: assets/styles/modules/_buttons.scss (hunk 7 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Done, I do not agree though, since the idea of making the animation parametric is to be able to have different instances of buttons with different behaviors. This idea also dies with the elimination of a parent class to include this mixin within.

File: assets/styles/modules/_buttons.scss (hunk 5 above)
Comment (ms-mrigoli) [reply assets/styles/modules/_buttons.scss]: Yes, my bad. Fixed, nice catch!

File: assets/styles/modules/_buttons.scss [hunk 8]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+    $sin45: 0.70710678118; //sqrt(2)/2 = sin(45 deg)
+    $cos45: $sin45;
+    $offset-y: ($width * $sin45 - $height + $stroke) / 2;
+    $offset-x: ($width * $cos45 - $width) / 2;
+    width: $width;
+    height: $height;
+    position: relative;
+    transform: rotate(0deg);
+    cursor: pointer;
+    span {
```
Comment (damianmuti) [inline assets/styles/modules/_buttons.scss]: Please check spacing of declarations.

File: assets/styles/modules/_buttons.scss [hunk 9]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+      opacity: 1;
+      left: 0;
+      transform: rotate(0deg);
+      transition: #{$mobile-menu-button-transition};
+      transform-origin: left center;
+
+      &:nth-child(1) {
+        top: 0;
+      }
+      &:nth-child(2) {
```
Comment (damianmuti) [inline assets/styles/modules/_buttons.scss]: Same about spacing.

Comment (damianmuti) [reply assets/styles/modules/_buttons.scss]: http://confluence.makingsense.com/display/UD/CSS+Development+styleguide#CSSDevelopmentstyleguide-codestructureandformatting

File: assets/styles/modules/_buttons.scss [hunk 10]
```diff
@@ -51,4 +51,78 @@ $c-button-border-radius: toem(5px) !default;
...
+    $cos45: $sin45;
+    $offset-y: ($width * $sin45 - $height + $stroke) / 2;
+    $offset-x: ($width * $cos45 - $width) / 2;
+    width: $width;
+    height: $height;
+    position: relative;
+    transform: rotate(0deg);
+    cursor: pointer;
+    span {
+      display: block;
```
Comment (damianmuti) [inline assets/styles/modules/_buttons.scss]: Check properties ordering.

http://confluence.makingsense.com/display/UD/CSS+Development+styleguide#CSSDevelopmentstyleguide-codestructureandformatting

File: assets/styles/templates/_home.scss [hunk 11]
```diff
@@ -2,4 +2,6 @@
 /// Home
 /// @group Home
 /// @author MS
-////
\ No newline at end of file
+////
+
+@include c-mobile-menu-button(23px, 14px, 2px, #424242);
```
Comment (damianmuti) [inline assets/styles/templates/_home.scss]: Mixin call should not be included in the PR since each module is not mandatory to be used.

File: assets/templates/index.html [hunk 12]
```diff
@@ -44,6 +44,12 @@
   <!-- /End of Header -->
   <!-- Content -->
   <main role="main">
+
+      <div class="mobile-menu-button">
```
Comment (damianmuti) [inline assets/templates/index.html]: Same goes with the HTML portion.

File: assets/styles/templates/_home.scss (hunk 11 above)
Comment (ms-mrigoli) [reply assets/styles/templates/_home.scss]: Yes, that was my bad, I was testing it and forgot to delete the call. Already fixed it.

File: assets/templates/index.html (hunk 12 above)
Comment (ms-mrigoli) [reply assets/templates/index.html]: This was unintended, The HTML was just for testing. Already deleted it.

**Comment by arzafran** [comment]
is there any reason why this is still not merged?

**Comment by ms-mrigoli** [comment]
I was waiting for @damianmuti to approve it.
